spec/support/helpers/request_spec_helpers.rb

```diff
@@ -50,4 +50,35 @@ def post_s(path, opts = {})
   def document_root_element
     html_document.root
   end
+
+  # Gets an OrderedHash of the following form for the descendants of a form.
+  def arrange_descendants(form)
+    sort = "(case when ancestry is null then 0 else 1 end), ancestry, rank"
+    # We eager load questions and option sets since they are likely to be needed.
+    nodes = form.descendants.includes(question: {option_set: :root_node}).order(sort).to_a
+    with_self = Questioning.arrange_nodes(nodes)
+    with_self.keys
+  end
+
+  # Arrange array of nodes into a nested hash of the form
+  # {node => children}, where children = {} if the node has no children
+  def arrange_nodes(nodes)
+    arranged = ActiveSupport::OrderedHash.new
+    min_depth = Float::INFINITY
+    index = Hash.new { |h, k| h[k] = ActiveSupport::OrderedHash.new }
+
+    nodes.each do |node|
+      children = index[node.id]
+      index[node.parent_id][node] = children
+
+      depth = node.depth
+      if depth < min_depth
+        min_depth = depth
+        arranged.clear
+      end
+      arranged[node] = children if depth == min_depth
+    end
+
+    arranged
+  end
```

Comment:
Can these helper methods please be moved to the `odk_submissions` context since that's the only place they're needed? And perhaps marked with comments to the effect that these methods should be removed when the spec is refactored to use the newer method of constructing XML fixtures.